baccmod/base_fit_acceptance_map_creator.py

```diff
@@ -0,0 +1,204 @@
+            interpolation_cleaning_spatial_relative_threshold=interpolation_cleaning_spatial_relative_threshold,
+        )
+
+        self.sq_rel_residuals = {"mean": [], "std": []}
+        self.name_normalisation_parameter = name_normalisation_parameter
+
+    @abstractmethod
+    def create_acceptance_map(self, observations):
+        """
+        Subclasses must implement: run self._create_base_computation_map(), then
+        call self.fit_background() for each energy slice, and finally pack into Background3D.
+        """
+        pass
+
+    def fit_background(self, model: Model, *coords: List[np.ndarray], count_map: np.ndarray, exp_map_total: np.ndarray, exp_map: np.ndarray) -> np.ndarray:
+        """
+        Perform a Poisson fit on the given map (could be 1D, 2D or 3), given
+        the fine‐binned exposure (exp_map) and total‐exposure (exp_map_total),
+        returning a smooth “counts model” on the same grid as `count_map`.
+
+        Parameters
+        ----------
+        model : astropy.modeling.model
+            the model to fit to the data
+        coords : List of np.ndarray
+            the list of coordinate
+        count_map : np.ndarray
+            counts in each pixel, integer
+        exp_map_total : np.ndarray
+            exposure WITHOUT exclusion regions
+        exp_map : np.ndarray
+            exposure CORRECTED for exclusion regions
+
+        Returns
+        -------
+        fitted_counts_model : 2D np.ndarray
+            The best‐fit model of counts (on the same pixel grid as count_map).
+        """
+
+        # Compute exposure correction
+        mask = exp_map_total > 0
+        exp_correction = exp_map / exp_map_total
+        exp_correction[~mask] = 0
+
+        # Initialise the model
+        model_init = model.copy()
+
+        # Correct normalisation of the model
+        if self.name_normalisation_parameter is not None:
+            # Compute correction
+            init_count_model = np.sum(model_init(*coords)*exp_correction)
+            correction_norm = np.sum(count_map)/init_count_model
+
+            # build list of free parameters
+            tied = model_init.tied or {}
+            all_params = list(model_init.param_names)
+            free_params = [p for p in all_params if (p not in tied) or (not tied[p]) and (not model_init.fixed.get(p, False))]
+
+            # Apply correction
+            for p in free_params:
+                if self.name_normalisation_parameter in p:
+                    setattr(model_init, p, getattr(model_init, p)*correction_norm)
+
+        # Fit the model
+        fitter = PoissonFitter()
+        best_model = fitter(model_init, *coords, data=count_map, exposure_correction=exp_correction)
+
+        # Collect results & (optionally) track residuals
+        if logger.getEffectiveLevel() <= logging.INFO:
+            fitted_model = best_model(*coords) * exp_correction
+            fitted_model[exp_map == 0] = 1.0
```

Comment:
if exp_map is zero, count_map should be zero. And the fitted model should be unconstrained by this pixels.

Here this adds -1 to the relative residuals for every pixels. THis could be kept as a trick to avoid dividing by zero but in this case masks will need to be implemented in later steps. I may have a fix localy on this issue, I will try to implement it this week.